### PR TITLE
Force active tag to the solution tag in MultiAppVariableValueSamplePostprocessorTransfer

### LINKED ISSUE #26684: MultiAppVariableValueSamplePostprocessorTransfer can attempt to access a ghost element of a non-ghosted tagged vector
## Bug Description
The call to `reinitElemPhys` in `MultiAppVariableValueSamplePostprocessorTransfer` can attempt to access a ghost element of a non-ghosted tagged vector if active vector tags are not set properly.

## Steps to Reproduce
Add a parallel tagged vector to the system and try to use `MultiAppVariableValueSamplePostprocessorTransfer`.

## Impact
Cannot add a parallel tagged vector when using `MultiAppVariableValueSamplePostprocessorTransfer`.

### PR DESCRIPTION
**Pull request by NamjaeChoi**
Closes #26684.

Related Griffin PR: https://github.inl.gov/ncrc/griffin/pull/1737.

### CODE REVIEW

**Comment by NamjaeChoi** [comment]
@YaqiWang Looks like you have contributed to this transfer before. Tagging you to see if you have any opinions.

**Comment by YaqiWang** [comment]
I think I was pathing this. I did something in Griffin related with this. I was adding
```
  auto & tags = getFEVariableCoupleableVectorTags();
  _fe_problem.setActiveFEVariableCoupleableVectorTags(tags, 0);
```
Not sure if we can do something similar here.

**Comment by NamjaeChoi** [comment]
> I think I was pathing this. I did something in Griffin related with this. I was adding
> 
> ```
>   auto & tags = getFEVariableCoupleableVectorTags();
>   _fe_problem.setActiveFEVariableCoupleableVectorTags(tags, 0);
> ```
> 
> Not sure if we can do something similar here.

Is transfer coupleable?

**Comment by YaqiWang** [comment]
Looks like no. But I would check how we get to the crash and see if we can loop through only solution vectors in use.

**Comment by NamjaeChoi** [comment]
> Looks like no. But I would check how we get to the crash and see if we can loop through only solution vectors in use.

I could not track down to the very end, but something seems to be happening in `MooseVariableData::computeValues`.

**Comment by YaqiWang** [comment]
> > Looks like no. But I would check how we get to the crash and see if we can loop through only solution vectors in use.
> 
> I could not track down to the very end, but something seems to be happening in `MooseVariableData::computeValues`.

Maybe somewhere this function is called `MooseVariableDataBase<OutputType>::setActiveTags` by passing all available tags.

**Comment by NamjaeChoi** [comment]
I printed the active tags and only two vectors were active: current solution and IQS residual. I think wherever we couple in the IQS residual, we should reset the active tag list once finished, but don't know how.

File: framework/src/transfers/MultiAppVariableValueSamplePostprocessorTransfer.C [hunk 1]
```diff
@@ -223,6 +223,8 @@ MultiAppVariableValueSamplePostprocessorTransfer::execute()
         mooseError("MultiAppVariableValueSamplePostprocessorTransfer does not support transfer of "
                    "vector variables");
 
+      _fe_problem.setActiveElementalMooseVariables({&_var}, 0);
```
Comment (GiudGiud) [inline framework/src/transfers/MultiAppVariableValueSamplePostprocessorTransfer.C:226]:
```suggestion
      _fe_problem.setActiveElementalMooseVariables({&_var}, /*thread_id=*/0);
```
same for the other

how do we know this variable is elemental?

Comment (NamjaeChoi) [reply framework/src/transfers/MultiAppVariableValueSamplePostprocessorTransfer.C:226]: Not familiar with other types of variables. Would adding `if (_var->isElemental())` work?

Comment (GiudGiud) [reply framework/src/transfers/MultiAppVariableValueSamplePostprocessorTransfer.C:226]: dont think we have that one. what are you trying to ban exactly here? variables missing ghosting?

Comment (NamjaeChoi) [reply framework/src/transfers/MultiAppVariableValueSamplePostprocessorTransfer.C:226]: Primarily yes, although this is not a perfect solution. How about `if (!_var->isNodal())`

Comment (GiudGiud) [reply framework/src/transfers/MultiAppVariableValueSamplePostprocessorTransfer.C:226]: will have to do. If we want a hasDofsOutsideNodes here

**Review by GiudGiud — APPROVED** [review]
fine to me as long as it passes tests

**Comment by NamjaeChoi** [comment]
How do you think about hard wiring `_fe_problem.setActiveFEVariableCoupleableVectorTags({MOOSE::SOLUTION_TAG}, 0);` in transfer?

**Comment by YaqiWang** [comment]
Or add a parameter to accept the vector tag name and default it to `{"solution"}`, which is the current solution vector.

**Comment by NamjaeChoi** [comment]
All the failures are coming from a test named `mixture_rayleigh_bernard_two_phase` and it is irrelevant with this PR.

**Comment by GiudGiud** [comment]
> Or add a parameter to accept the vector tag name and default it to {"solution"}

Only if anyone actually needs this. Micro-optimizing transfers penalizes maintenance.

**Review by lindsayad — COMMENTED** [review]
this seems like a patch for a potentially larger issue. I won't be that surprised if we see something similar crop up again

**Comment by YaqiWang** [comment]
Typically loops call `FEProblemBase::setActiveFEVariableCoupleableVectorTags` so variables can grab active solution vectors. But here the element loop does not have that call before this change and is using FEProblemBase reinit. It would be nice that `_fe_problem.reinitElemPhys` can hit an error or assertion to remind us to call `setActiveFEVariableCoupleableVectorTags`.

**Comment by NamjaeChoi** [comment]
> this seems like a patch for a potentially larger issue. I won't be that surprised if we see something similar crop up again

Yeah, I felt the same way. What could be a fundamental resolution?

**Comment by lindsayad** [comment]
I thing my comment was wrong. @YaqiWang is right that objects doing calls to things like `_fe_problem.reinitXXX` have calls to things like `_fe_problem.setActiveFEVariableCoupleableVectorTags` out in front of them. More error checking or adding new APIs that group some of these functions might be good

**Pull request merged**